src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
 
+    public MateStack() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (elementData.length == size) {
+            grow();
+        }
+        elementData[size++] = value;
     }
 
     public T peek() {
-        return null;
+        empty();
+        return elementData[size - 1];
     }
 
     public T pop() {
-        return null;
+        empty();
+        T removedElement = peek();
+        elementData[size--] = null;
+        return removedElement;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    public boolean empty() {
```

Comment:
You can make this method void

Reply:
I tried to override default empty() method, just like the rest) so it must be public boolean...